### PR TITLE
Caching: SE.Redis update and fix naming inconsistency

### PR DESCRIPTION
**Pull request by mgravell**
Resolve redis cache naming inconsistencies

The cache intent is included in the connection metadata ... or should be.

- rev SE.Redis; among other fixes, this resolves an issue with some client identifier strings being ignored at the server
- consistently apply identifier in cache setup
  - this was applied in async connect, but not sync connect
  - this was not applied if the factory callback was used (Aspire uses this)
- avoid cloning the configuration options (breaks key-rotation scenarios)
- adds (opt-in) RESP3 support (via lib-rev), and assorted lib fixes

manual test evidence (from running tests with a pause in):

``` txt
> redis-cli client list
...
id=348 ... resp=2 lib-name=SE.Redis-aspnet-DC lib-ver=2.7.27.49176
id=345 ... resp=2 lib-name=SE.Redis-aspnet-DC lib-ver=2.7.27.49176
```

### CODE REVIEW

File: eng/Baseline.Designer.props [hunk 1]
```diff
@@ -374,15 +374,15 @@
   </PropertyGroup>
   <ItemGroup Condition=" '$(PackageId)' == 'Microsoft.AspNetCore.DataProtection.StackExchangeRedis' AND '$(TargetFramework)' == 'net462' ">
     <BaselinePackageReference Include="Microsoft.AspNetCore.DataProtection" Version="[8.0.0, )" />
-    <BaselinePackageReference Include="StackExchange.Redis" Version="[2.6.122, )" />
+    <BaselinePackageReference Include="StackExchange.Redis" Version="[2.7.23, )" />
```
Comment (wtgodbe) [inline eng/Baseline.Designer.props:377]: Was this updated by a repo-wide find/replace? If so please revert, this file should only be updated after a release

Comment (mgravell) [reply eng/Baseline.Designer.props:377]: Yes, my bad; ctrl-f misunderstanding - will revert

**Review by wtgodbe — APPROVED** [review]

**Comment by mgravell** [comment]
SignalR  (test fail) should just be some `using` directive; will look at tomorrow

File: src/Caching/StackExchangeRedis/src/RedisCacheOptions.cs [hunk 2]
```diff
@@ -59,18 +59,13 @@ internal bool UseForceReconnect
         set => _useForceReconnect = value;
     }
 
-    internal ConfigurationOptions GetConfiguredOptions(string libSuffix)
+    internal ConfigurationOptions GetConfiguredOptions()
     {
-        var options = ConfigurationOptions?.Clone() ?? ConfigurationOptions.Parse(Configuration!);
```
[the remark this answers is not in the document]
Comment (mgravell) [reply src/Caching/StackExchangeRedis/src/RedisCacheOptions.cs:64]: @BrennanConroy  the thing potentially it breaks can fundamentally cause access to the servers to be lost (key rotation for managed identity); let me review the azure redis documentation to see if we can find a compromise here

[the remark this answers is not in the document]
Comment (mgravell) [reply src/Caching/StackExchangeRedis/src/RedisCacheOptions.cs:64]: @BrennanConroy this `Clone()` is a [recent addition](https://github.com/dotnet/aspnetcore/commit/792f1f218582f2c5cf21fd6379832f85d9e7c065#diff-747e1ff20f62e31c87fa0d9ac53821ca44c24ba962a789def6447151aee57926L312); I'm not concerned by sharing the connect-fail change here, so IMO we're *restoring* the correct behaviour here;

File: src/Caching/StackExchangeRedis/src/RedisCache.cs [hunk 3]
```diff
@@ -369,6 +363,19 @@ private void TryRegisterProfiler(IConnectionMultiplexer connection)
...
 
+    private static void TryAddSuffix(IConnectionMultiplexer connection)
+    {
+        try
+        {
+            connection.AddLibraryNameSuffix("aspnet");
+            connection.AddLibraryNameSuffix("DC");
+        }
+        catch (Exception ex)
+        {   // not critical
```
[the remark this answers is not in the document]
Comment (mgravell) [reply src/Caching/StackExchangeRedis/src/RedisCache.cs]: resolved

File: src/Caching/StackExchangeRedis/src/RedisCache.cs [hunk 4]
```diff
@@ -369,6 +363,19 @@ private void TryRegisterProfiler(IConnectionMultiplexer connection)
...
+            Debug.WriteLine(ex.Message);
```
[the remark this answers is not in the document]
Comment (mgravell) [reply src/Caching/StackExchangeRedis/src/RedisCache.cs]: resolved

File: src/Middleware/Microsoft.AspNetCore.OutputCaching.StackExchangeRedis/src/RedisOutputCacheStore.Log.cs [hunk 5]
```diff
@@ -14,4 +14,7 @@ internal partial class RedisOutputCacheStore
 
     [LoggerMessage(2, LogLevel.Error, "Fatal error occurred executing redis output-cache GC loop.", EventName = "RedisOutputCacheGCFatalError")]
     internal static partial void RedisOutputCacheGCFatalError(ILogger logger, Exception exception);
+
+    [LoggerMessage(3, LogLevel.Debug, "Unable to add library name suffix.", EventName = nameof(UnableToAddLibraryNameSuffix))]
```
Comment (BrennanConroy) [inline src/Middleware/Microsoft.AspNetCore.OutputCaching.StackExchangeRedis/src/RedisOutputCacheStore.Log.cs:18]: nit: Don't use nameof for event name, we don't want to accidentally change the event name if we rename the method.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**